### ISSUE TITLE
Nautobot 2.0: ScheduledJob migration KeyError: 'commit'

### ISSUE THREAD

**Issue by Thetacz** [nautobot/nautobot#3889]
### Environment
* Nautobot version (Docker tag too if applicable): 2.0.0-beta.1
* Python version: 3.10
* Database platform, version: PostreSQL
* Middleware(s): -

### Steps to Reproduce
1. Create ScheduledJob
2. Upgrade Nautobot from 1.5.19 to 2.0.0-beta.1

See Slack thread https://networktocode.slack.com/archives/C01NWPK6WHL/p1686297326868139 for more context and reason why this happens.

### Expected Behavior
migration `extras/0090_scheduledjob__data_migration` passes

### Observed Behavior
```
Running migrations:
  Applying extras.0090_scheduledjob__data_migration...Traceback (most recent call last):
  File "/usr/local/bin/nautobot-server", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/nautobot/core/cli/__init__.py", line 54, in main
    run_app(
  File "/usr/local/lib/python3.10/site-packages/nautobot/core/runner/runner.py", line 263, in run_app
    management.execute_from_command_line([runner_name, command] + command_args)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.10/site-packages/nautobot/core/management/commands/post_upgrade.py", line 78, in handle
    call_command(
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 181, in call_command
    return command.execute(*args, **defaults)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/commands/migrate.py", line 244, in handle
    post_migrate_state = executor.migrate(
  File "/usr/local/lib/python3.10/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.10/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.10/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.10/site-packages/django/db/migrations/migration.py", line 126, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.10/site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/usr/local/lib/python3.10/site-packages/nautobot/extras/migrations/0090_scheduledjob__data_migration.py", line 19, in migrate_existing_scheduled_jobs
    del sj.kwargs["commit"]
KeyError: 'commit'
```

**Comment by glennmatthews** [comment]
Looks to be bad logic in the migration - I would expect `commit` to be a key in `old_kwargs`, not a key in `old_kwargs["data"]`.

**Comment by HanlinMiao** [comment]
closed by #3907
